app.js

```diff
@@ -276,6 +276,13 @@ models.sequelize.authenticate().then(function () {
     } else {
       throw new Error('server still not ready after db synced')
     }
+    // Allow to timeout-lock guest notes so they can no longer be changed after the timeout period
+    if (config.lockGuestNotes.timeout > 0) {
+      setInterval(models.Note.lockOldGuestNotes, (config.lockGuestNotes.timeout / 2) * 1000, config.lockGuestNotes.timeout, config.lockGuestNotes.permission)
```

Comment:
Why don't you check the database every e.g. 60 minutes? Otherwise the lock might happen waaaay to late.